### DISCUSSION TITLE
Farcaster docs outdated/tutorial not working

### DISCUSSION THREAD

**Discussion by ODOlarte** [farcasterxyz/protocol#48] [Q&A] [answered]
Hello everyone!

I recently started to play around with Farcaster and I tried to run the tutorial that you have to fetch casts, and it didn't work even when I tried with Goerli. I received this error every time (I also tried with Replit).

`node:1636) UnhandledPromiseRejectionWarning: Error: missing revert data in call exception (error={"reason":"processing response error","code":"SERVER_ERROR","body":"{\"jsonrpc\":\"2.0\",\"id\":43,\"error\":{\"code\":-32000,\"message\":\"execution reverted\"}}","error":{"code":-32000},"requestBody":"{\"method\":\"eth_call\",\"params\":[{\"to\":\"0xe3be01d99baa8db9905b33a3ca391238234b79d1\",\"data\":\"0x342740c97600000000000000000000000000000000000000000000000000000000000000\"},\"latest\"],\"id\":43,\"jsonrpc\":\"2.0\"}","requestMethod":"POST","url":"https://eth-goerli.alchemyapi.io/v2/<api_key>}, data="0x", code=CALL_EXCEPTION, version=providers/5.5.2)`

Not sure if this is the right place to mention this, just want to let you know about it and if you have a solution for it, I would appreciate that.

Let me know if you have any questions

Thanks,

**Comment by varunsrin** [comment, marked as answer]
The contracts have been changed recently and the tutorial is out of date. Please see the repo for the new contract signatures: https://github.com/farcasterxyz/contracts, a tutorial will be added soon
